modules/tools/gen_vehicle_protocol/gen_protocols.py

```diff
@@ -66,7 +66,11 @@ def gen_report_cpp(car_type, p, output_dir):
         fmt_val["protocol_name_lower"] = p["name"]
         classname = p["name"].replace('_', '').capitalize()
         fmt_val["classname"] = classname
-        fmt_val["id_upper"] = p["id"].upper()
+        a = int(p["id"].upper(),16)
```

Comment:
1. Space after `,`
2. Avoid single-char variable names. Please also fix the `p`.